Automattic/jetpack
# Jetpack Blocks: Set default editor style to be registered along with blocks if no editor styles are provided

This change ensures that the default Jetpack editor styles are added as an `editor_style` dependency when registering Jetpack blocks within the Jetpack plugin. In the new FSE site editor , where the editor canvas is rendered within an iframe, styles are copied to the iframe based on whether they're registered against the block ([here](https://github.com/WordPress/gutenberg/blob/9484b1ed1dd60ad9cbc721e91e35f4b7b5ed7143/lib/client-assets.php#L698)'s where the logic happens).

Without this change, when the Jetpack styles are loaded via an external link (as they are in wpcom) they're not picked up in the fallback [styleSheetsCompat](https://github.com/WordPress/gutenberg/blob/748421d1980b1aeb486283d90d09a9bfefde068d/packages/block-editor/src/components/iframe/index.js#L32) call, and we're left with unstyled blocks in the editor. For example:

### Unstyled blocks in the new site editor

| Donation | Business Hours |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/111581881-26fd8280-880e-11eb-8d2c-e17f50d8dd8b.png) | ![image](https://user-images.githubusercontent.com/14988353/111581899-2b29a000-880e-11eb-8c6b-537dc42f2c04.png) |
| Cover | Pay with Paypal |
| ![image](https://user-images.githubusercontent.com/14988353/111581914-31b81780-880e-11eb-8dc2-c7f5edc9c2be.png) | ![image](https://user-images.githubusercontent.com/14988353/111581926-35e43500-880e-11eb-9538-f5da052c5aec.png) |

### After (with this change applied)

| Donation | Business Hours |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/111582171-8fe4fa80-880e-11eb-95d7-1cddbe1c1a7f.png) | ![image](https://user-images.githubusercontent.com/14988353/111582202-9c695300-880e-11eb-9d3b-77b530d7931d.png) |
| Cover | Pay with Paypal |
| ![image](https://user-images.githubusercontent.com/14988353/111582250-ac813280-880e-11eb-9f5f-cbafd25a314c.png) | ![image](https://user-images.githubusercontent.com/14988353/111582536-126dba00-880f-11eb-8325-417b4534548b.png) |

Related to https://github.com/Automattic/wp-calypso/issues/50407

#### Changes proposed in this Pull Request:

* When running in a Jetpack plugin context (the `Jetpack_Gutenberg` class is available):
  * If there is no `editor_style` set for the block, set it to the default editor styles that are registered by the plugin.
  * If there is an `editor_style` set for the block, don't change it.
* When running outside of a Jetpack plugin context (the `Jetpack_Gutenberg` class is not available, then don't make any changes to the `editor_style`.
* Add an `is_standalone_block` method to abstract away the `Jetpack_Gutenberg` class check.
* Add tests to cover this behaviour.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






At the moment, I can test this change in my sandbox by manually copying the changes over to the `mu-plugins/jetpack-plugin/vendor/automattic/jetpack-blocks/src/` directory. To do so you might use a command like: `scp ~/dev/a8c-projects/jetpack/projects/packages/blocks/src/class-blocks.php wpcom:/home/wpcom/public_html/wp-content/mu-plugins/jetpack-plugin/vendor/automattic/jetpack-blocks/src/class-blocks.php`.

1. Create a test FSE site at https://horizon.wordpress.com/new?flags=gutenboarding/site-editor
2. In your `0-sandbox` file set the following and sandbox s0.wp.com, s1.wp.com, s2.wp.com:

```
add_filter( 'wpcom_sandbox_staticize_domain', '__return_true' );
```

More context in p7H4VZ-2qv-p2

3. Open up the site editor (the FSE editor, e.g. `https://wordpress.com/site-editor/mytestsitedomain.com`) and try adding the Donation, Business Hours, or Pay with Paypal blocks and note that the styling isn't being loaded correctly. If you add the Cover block, you'll see the upgrade nudge, even though it shouldn't be visible.
4. Manually update the `class-blocks.php` file as in this PR and ensure your test site is sandboxed.
5. Reload the editor, and the styling should be fixed.
